src/osd/osd_types.h

```diff
@@ -4383,7 +4382,28 @@ struct pg_log_entry_t {
   void decode(ceph::buffer::list::const_iterator &bl);
   void dump(ceph::Formatter *f) const;
   static void generate_test_instances(std::list<pg_log_entry_t*>& o);
-
+  
+  size_t encode_size() const {
+    size_t snaps_size = 0;
+    for (unsigned i = 0; i < snaps.length(); i++) {
+      snaps_size += sizeof(snaps[i]);
+    }
```

Comment:
I don't think this size estimate is correct.  sizeof gives the static, in-memory size of the struct.  Can we warn based on the actual length of the bufferlist after encode?

Reply:
Alternately, converting pg_log_entry_t to use denc?